src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java

```diff
@@ -1615,4 +1614,144 @@ private void addRequestTypeInProfileWF(WfRequest wfRequest){
 				wfRequest.setRequestType(requestKey);
 		}
 	}
+
+	public Response workflowTransitionV1(String rootOrg, String org, WfRequest wfRequest, String userId, String role) {
+		HashMap<String, String> changeStatusResponse;
+		List<String> wfIds = new ArrayList<>();
+		String changedStatus = null;
+		Response response = new Response();
+		HashMap<String, Object> data = new HashMap<>();
+
+		if (!CollectionUtils.isEmpty(wfRequest.getUpdateFieldValues())) {
+			String wfId = wfRequest.getWfId();
+			for (HashMap<String, Object> updatedField : wfRequest.getUpdateFieldValues()) {
+				wfRequest.setUpdateFieldValues(wfRequest.getUpdateFieldValues());
+				wfRequest.setWfId(wfId);
+				if (!StringUtils.isEmpty(wfRequest.getServiceName())) {
+					if (StringUtils.isEmpty(wfRequest.getWfId()) && wfRequest.getServiceName().equalsIgnoreCase(Constants.PROFILE_SERVICE_NAME)) {
+						if (handleProfileServiceWorkflowV1(wfRequest, response, data)) {
+							return response;
+						}
+					}
+				}
+
+				changeStatusResponse = changeStatus(rootOrg, org, wfRequest, userId, role);
+				wfIds.add(changeStatusResponse.get(Constants.WF_ID_CONSTANT));
+				changedStatus = changeStatusResponse.get(Constants.STATUS);
+			}
+		} else {
+			if (!StringUtils.isEmpty(wfRequest.getServiceName())) {
+				if (StringUtils.isEmpty(wfRequest.getWfId()) && wfRequest.getServiceName().equalsIgnoreCase(Constants.PROFILE_SERVICE_NAME)) {
+					if (handleProfileServiceWorkflowV1(wfRequest, response, data)) {
+						return response;
+					}
+				}
+			}
+
+			changeStatusResponse = changeStatus(rootOrg, org, wfRequest, userId, role);
+			wfIds.add(changeStatusResponse.get(Constants.WF_ID_CONSTANT));
+			changedStatus = changeStatusResponse.get(Constants.STATUS);
+		}
+
+		// Populate the response
+		data.put(Constants.STATUS, changedStatus);
+		data.put(Constants.WF_IDS_CONSTANT, wfIds);
+		response.put(Constants.MESSAGE, Constants.STATUS_CHANGE_MESSAGE + changedStatus);
+		response.put(Constants.DATA, data);
+		response.put(Constants.STATUS, HttpStatus.OK);
+		return response;
+	}
+
+
+	private boolean handleProfileServiceWorkflowV1(WfRequest wfRequest, Response response, Map<String, Object> data) {
```

Comment:
this is returning true always... so remove the return type... make sure response object updated with details and calling method is using the response object properly...